android/src/main/java/com/brentvatne/exoplayer/FullScreenPlayerView.kt

```diff
@@ -127,4 +134,29 @@ class FullScreenPlayerView(
         layoutParams.setMargins(0, 0, 0, 0)
         return layoutParams
     }
+
+    private fun navigationBarVisibility() {
+        if (controlsConfig.hideNavigationBarOnFullScreenMode) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                window?.setDecorFitsSystemWindows(false)
+                window?.insetsController?.let {
+                    it.hide(WindowInsets.Type.navigationBars())
+                    it.systemBarsBehavior = WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+                }
+            } else {
+                @Suppress("DEPRECATION")
+                window?.decorView?.systemUiVisibility = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+            }
+        } else {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                window?.setDecorFitsSystemWindows(true)
+                window?.insetsController?.let {
+                    it.show(WindowInsets.Type.navigationBars())
+                }
+            } else {
+                @Suppress("DEPRECATION")
+                window?.decorView?.systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+            }
+        }
```

Comment:
You can move `if (controlsConfig.hideNavigationBarOnFullScreenMode) {` inside window?.insetsController?.let, then we avoid repeating ->
```tsx
if(Build.VERSION.SDK_INT>=Build.VERSION_CODES.R){
    window?.setDecorFitsSystemWindows(false)
    window?.insetsController?.let{
        if(controlsConfig.hideNavigationBarOnFullScreenMode){
            it.hide(WindowInsets.Type.navigationBars())
            it.systemBarsBehavior=WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        }else{
            it.show(WindowInsets.Type.navigationBars())
        }
    }
}else{
    @Suppress("DEPRECATION")
    window?.decorView?.systemUiVisibility=View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
}
```